### PR TITLE
feat: update xcm config

### PR DESCRIPTION
**Pull request by Roznovjak**
Based on a discussion with @apopiak, where we reviewed the XCM changes proposed by the Substrate builders programme in this PR #591, we decided to make the following changes:
1. Use `WithComputedOrigin` derivative barrier with `AllowTopLevelPaidExecutionFrom` and `AllowSubscriptionsFrom` barriers. This outer barrier computes the effective origin (checks for any origin altering operations) and passes this updated origin to inner barriers.
2. Change `XcmExecuteFilter` from `Everything` to `Nothing`. This change disallows `pallet_xcm::execute` extrinsic which allows sending arbitrary XCM messages. We are not aware of any issues with this param set to `Everything`, but better be sure. Moreover, `pallet_xcm` is blocked by the runtime call filter.

### CODE REVIEW

**Comment by enthusiastmartin** [comment]
Can you add description why the computedorigin is required ?

File: runtime/hydradx/src/xcm.rs [hunk 1]
```diff
@@ -192,7 +199,7 @@ impl pallet_xcm::Config for Runtime {
 	type SendXcmOrigin = EnsureXcmOrigin<RuntimeOrigin, LocalOriginToLocation>;
 	type XcmRouter = XcmRouter;
 	type ExecuteXcmOrigin = EnsureXcmOrigin<RuntimeOrigin, LocalOriginToLocation>;
-	type XcmExecuteFilter = Everything;
+	type XcmExecuteFilter = Nothing;
```
Comment (mrq1911) [inline runtime/hydradx/src/xcm.rs:202]: @apopiak @dmoka this effectively breaks the exchange and send back assets scenario correct?

Comment (mrq1911) [reply runtime/hydradx/src/xcm.rs:202]: if this is the case i would reconsider this change

Comment (apopiak) [reply runtime/hydradx/src/xcm.rs:202]: I would say it makes sense to define the more permissive filter in the context of the ExchangeAsset stuff, not here.

File: runtime/hydradx/src/xcm.rs [hunk 2]
```diff
@@ -35,11 +35,18 @@ pub type LocalOriginToLocation = SignedToAccountId32<RuntimeOrigin, AccountId, R
...
-	AllowSubscriptionsFrom<Everything>,
+	// Evaluate the barriers with the effective origin
+	WithComputedOrigin<
+		(
+			AllowTopLevelPaidExecutionFrom<Everything>,
+			// Subscriptions for version tracking are OK.
+			AllowSubscriptionsFrom<Everything>,
+		),
+		UniversalLocation,
+		ConstU32<1>,
```
Comment (apopiak) [inline runtime/hydradx/src/xcm.rs:48]: Seems like 1 should be fine, but I don't see a reason to deviate from the ecosystem which is going for 8.

**Review by apopiak — APPROVED** [review]
looks fine after adjusting computed origin count

File: runtime/hydradx/src/xcm.rs (hunk 2 above)
Comment (apopiak) [inline runtime/hydradx/src/xcm.rs:48]:
```suggestion
		ConstU32<8>,
```

**Pull request merged**